### PR TITLE
(REVIEW ONLY, DON'T MERGE)ENG-14901, track client-side backpressure on every connection.

### PR DESCRIPTION
**Pull request by luyangco**
Change-Id: I16cf1b8a99b760d1b03d4e30c85bf159a88c0b35

Pro: https://github.com/VoltDB/pro/pull/2391

### CODE REVIEW

File: src/frontend/org/voltdb/client/ClientImpl.java [hunk 1]
```diff
@@ -644,37 +655,44 @@ public void close() throws InterruptedException {
 
     @Override
     public void backpressureBarrier() throws InterruptedException {
-        backpressureBarrier( 0, 0);
+        backpressureBarrier(new PerConnectionBackpressure(null, false), 0, 0);
```
Comment (zeemanhe) [inline src/frontend/org/voltdb/client/ClientImpl.java]: **pass a null connectionId** violates the assertion in the constructor.

File: src/frontend/org/voltdb/client/ClientImpl.java [hunk 2]
```diff
@@ -525,6 +530,8 @@ private final boolean internalAsyncCallProcedure(
             catch (InterruptedException e) {
                 throw new java.io.InterruptedIOException("Interrupted while invoking procedure asynchronously");
             }
+            // requeue the invocation once backpressure is ceased
```
Comment (rdykiel) [inline src/frontend/org/voltdb/client/ClientImpl.java]: "has" ceased

File: src/frontend/org/voltdb/client/Distributer.java [hunk 3]
```diff
@@ -1198,6 +1213,12 @@ boolean queue(
                             if (!cxn.hadBackPressure() || ignoreBackpressure) {
                                 backpressure = false;
                             }
+                        } else {
+                            // For FAST reads, choose master if no viable replica found
+                            cxn = m_partitionMasters.get(hashedPartition);
+                            if (cxn != null && !cxn.hadBackPressure() || ignoreBackpressure) {
```
Comment (zeemanhe) [inline src/frontend/org/voltdb/client/Distributer.java]: should find a connection without back pressure even if ignoreBackpressure== true?

**Comment by rdykiel** [comment]
I'm ok with these changes

**Comment by luyangco** [comment]
Not going to merge any time soon.

**Pull request closed without merging**